src/eval_cuda.cpp

```diff
@@ -321,31 +321,37 @@ void jitc_assemble_cuda_func(const char *name, uint32_t inst_id,
                            "    setp.ne.u16 %%p%u, %%w0, 0;\n",
                            v->param_offset, v->reg_index);
             }
-        } else if (v->data || vt == VarType::Pointer) {
+        } else if (v->literal || v->data) {
```

Comment:
I am confused: do we want to take this branch this for all literals? I would assume that most literals still go through the ``jitc_render_stmt_cuda`` or ``jitc_render_stmt_llvm`` functions. It's only the literals that were created before recording the vcall that should be opaque-ified in this way. Maybe I am misunderstanding.

Reply:
We indeed take this branch for all literals. At this point we can't use the heuristic based on CSE scope (or now indices order) to determine whether this branch should be taken or not. But we then check the offset in the datamap, which will be `-1` for all literal values that can be baked in the kernel.